src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+    if (func_8004356C(&this->dyna) != 0) {
+        this->unk_16A = 0x14;
+        this->dyna.actor.world.rot.y = Camera_GetCamDirYaw(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x4000;
+        this->actionFunc = func_8088B69C;
+    }
+}
+
+void func_8088B69C(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (this->unk_16A != 0) {
+        this->unk_16A--;
+    }
+
+    if (this->unk_16A != 0) {
+        this->dyna.actor.world.pos.x =
+            this->dyna.actor.home.pos.x + 5.0f * Math_SinS((this->dyna.actor.world.rot.y + (this->unk_16A << 0xE)));
+        this->dyna.actor.world.pos.z =
+            this->dyna.actor.home.pos.z + 5.0f * Math_CosS((this->dyna.actor.world.rot.y + (this->unk_16A << 0xE)));
+    } else {
+        this->dyna.actor.world.pos.x = this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z;
+        func_8088B24C(this);
+    }
+
+    if (!(this->unk_16A % 4)) {
+        func_800AA000(this->dyna.actor.xyzDistToPlayerSq, 0xB4, 0x0A, 0x64);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_SHAKE);
+    }
+}
+
+void func_8088B79C(BgHidanRock* this, GlobalContext* globalCtx) {
+    this->unk_16A--;
+    if ((this->dyna.actor.bgCheckFlags & 2) != 0) {
+        if (this->params == 0) {
+            this->unk_16A = 0x3C;
+            this->actionFunc = func_8088B5F4;
+        } else {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 15.0f;
+            this->actionFunc = func_8088B90C;
+            this->dyna.actor.flags &= ~0x30;
+        }
+
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        Audio_PlayActorSound2(
+            &this->dyna.actor,
+            (SurfaceType_GetSfx(&globalCtx->colCtx, this->dyna.actor.floorPoly, this->dyna.actor.floorBgId) + 0x800) &
+                0xFFFF);
+    }
+
+    this->unk_16C -= 0.5f;
+    this->unk_16C = (this->unk_16C < 0.0f) ? (0.0f) : this->unk_16C;
+
+    if (this->params == 0) {
+        if (func_8004356C(&this->dyna) != 0) {
+            if (this->unk_169 == 0) {
+                this->unk_169 = 3;
+            }
+            Camera_ChangeSetting(globalCtx->cameraPtrs[0], 0x30);
+        } else if (func_8004356C(&this->dyna) == 0) {
+            if (this->unk_169 != 0) {
+                Camera_ChangeSetting(globalCtx->cameraPtrs[0], 3);
+            }
+            this->unk_169 = 0;
+        }
+    }
+}
+
+void func_8088B90C(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 1.0f) != 0) {
+        this->actionFunc = func_8088B634;
+    }
+}
+
+void func_8088B954(BgHidanRock* this, GlobalContext* globalCtx) {
+    s16 unk_16A;
+
+    unk_16A = this->unk_16A;
+    if (unk_16A != 0) {
+        this->unk_16A = unk_16A - 1;
+        unk_16A = this->unk_16A;
+    }
+
+    if (unk_16A == 0) {
+        this->actionFunc = func_8088B79C;
+        this->dyna.actor.velocity.y = 0.0f;
+    }
```

Comment:
```suggestion
    if (unk_16A != 0) {
        this->unk_16A--;
    }

    if (this->unk_16A == 0) {
        this->actionFunc = func_8088B79C;
        this->dyna.actor.velocity.y = 0.0f;
    }
```
same question as before.